### DISCUSSION TITLE
How to get likes relation using pocketbase

### DISCUSSION THREAD

**Discussion by heyarviind** [pocketbase/pocketbase#1589] [Q&A] [answered]
Hi Everyone,

I have 2 collections `posts` and `likes`, whenever someone likes the post I add that to the likes table with `postId, userId`. 

How do I retrieve posts with the current user info if they have liked the post or not (Instead of running another query for each post)?

Thank you

**Comment by ganigeorgiev** [comment, marked as answer]
You can use indirect expand - https://pocketbase.io/docs/expanding-relations/#indirect-expand.

In your case you can retrieve the `posts` with all the users related from the `likes` collection like this:

```js
const result = await pb.collection('posts').getList(1, 30, {
    // or just "likes(postId)" if you need only access to the user id from the relation
    expand: "likes(postId).userId"
})
```

Note that your request client will be able to expand only the rel records to which they have View API rule access (in the above example for both the `likes` and `users` collections).

**Reply by heyarviind** [reply]
thanks this worked 🎉

**Reply by heyarviind** [reply]
I have another user case where I am fetching a list of posts. I want to show to the logged in user if they liked that post. I want to include that logic in the same query to check if that loggedin user liked the product or not.

**Reply by ganigeorgiev** [reply]
@heyarviind Please don't spam old discussions and instead open a new one elaborating more clearly on the actual problem and your collections structure (you can export it as json from the Dashboard > Settings > Export collections). See also [XY problem](https://xyproblem.info/).
